src/skirmishbunker/FlatRoof.py

```diff
@@ -1,59 +1,338 @@
+            .slot2D(slot_length_md, slot_radius)
+            .extrude(self.tile_height * 2)
+            .rotate((0, 0, 1), (0, 0, 0), 45)
+            .translate((0 - slot_translate, 0 - slot_translate, 0 - (self.tile_height / 2))))
+
+        slot3 = (cq.Workplane("XY")
+            .slot2D(slot_length_md, slot_radius)
+            .extrude(self.tile_height * 2)
+            .rotate((0, 0, 1), (0, 0, 0), 45)
+            .translate((slot_translate, slot_translate, 0 - (self.tile_height / 2))))
+
+        slot4 = (cq.Workplane("XY")
+            .slot2D(slot_length_sm, slot_radius)
+            .extrude(self.tile_height * 2)
+            .rotate((0, 0, 1), (0, 0, 0), 45)
+            .translate((0 - (slot_translate *2), 0 - (slot_translate * 2), 0 - (self.tile_height / 2))))
+
+        slot5 = (cq.Workplane("XY")
+            .slot2D(slot_length_sm, slot_radius)
+            .extrude(self.tile_height * 2)
+            .rotate((0, 0, 1), (0, 0, 0), 45)
+            .translate((slot_translate * 2, slot_translate * 2, 0 - (self.tile_height / 2))))
+
+        tile = (tile
+                .cut(slot)
+                .cut(slot2)
+                .cut(slot3)
+                .cut(slot4)
+                .cut(slot5))
+
+        columns = math_floor(width / (tile_space))
+        rows = math_floor(length / (tile_space))
+        tile_grid = grid.make_grid(
+            part = tile,
+            dim = [tile_space, tile_space],
+            columns = columns,
+            rows = rows
+        )
+
+        self.tiles = tile_grid.translate((0, 0, self.calc_tile_z_translate()))
+
+    def make_hatches(self):
+        int_length = self.calc_hatch_space_length() 
+        length_offset = self.calc_hatch_length_offset()
+        int_width = self.calc_hatch_space_width()
+        z_translate = self.calc_hatch_z_translate()
+
+        bp = Hatch()
+        bp.length = self.hatch_length
+        bp.width = self.hatch_width
+        bp.height = self.hatch_height
+        bp.hatch_radius = self.hatch_radius
+        bp.make()
+
+        print(bp.length)
+        print(bp.width)
+        print(bp.height)
+        print(z_translate)
+
+        hatch = bp.build()
+
+        series = SeriesHelper()
+        series.shape = hatch
+        series.outer_length = int_length
+        series.outer_width = int_width
+        series.length_offset = length_offset
+        series.comp_length = self.panel_length
+        series.comp_padding = self.panel_padding
+        series.x_translate = (int_length / 2) - (bp.width / 2)
+        series.y_translate = (int_width / 2) - (bp.width / 2)
+        series.z_translate = z_translate
+        series.keep_list = self.hatch_panels
+        series.make()
+
+        self.hatches = series.get_scene()
+
+    def make_hole_cuts(self):
+        x_translate = self.calc_hole_x_translate()
+        y_translate = self.calc_hole_y_translate()
+        z_translate = self.calc_hole_z_translate()
+
+        hole = cq.Workplane("XY").cylinder(
```

Comment:
@mlewis-everley fun quirk cylinder method signature is height then radius. I believe this led to a couple bugs